Fix pending checks not always rendering as pending

mskelton opened the pull request:
# Summary

Pending checks that are defined in the `StatusContext` were not properly rendering as pending.  Check runs worked fine, but external CI services like Buildkite which use `StatusContext` were not rendering properly.

I also improved the formatting of the pending checks pill which was rendering strangely before.

## How did you test this change?

Verified that pending checks work properly after this change.

## Images/Videos

### Before
![image](https://user-images.githubusercontent.com/25914066/170786017-5a598954-58f3-47f9-a93a-715ee6c10eae.png)

### After

![image](https://user-images.githubusercontent.com/25914066/170785607-d9a941c5-d938-4ef8-9150-046a9d27ef4d.png)

dlvhdr commented:
Looks good overall @mskelton!
I checked this locally and for me the checks are now incorrect.
At Wix, where I work, our CI probably reports things differently.

![image](https://user-images.githubusercontent.com/6196971/170818618-196a8329-26e1-4811-88a5-57e835d42611.png)

As you can see I have a green check, but the pill shows in gray.

mskelton commented:
Oops, forgot a hard coded conditional I changed for testing.  Fixed in https://github.com/dlvhdr/gh-dash/pull/134/commits/58a3563f1829482122812da3759bae874b5cddc6.

dlvhdr commented on ui/components/prsidebar/prsidebar.go:
```diff
@@ -98,7 +97,9 @@ func (m *Model) renderChecksPill() string {
-			Render(constants.WaitingGlyph + " Checks")
+			Foreground(styles.DefaultTheme.MainText).
+			Faint(true).
+			Render(" Checks")
```
Can you set the waiting glyph in the constants pkg? Instead of hardcoding it here

mskelton replied:
The reason I hard coded it here was to fix it not displaying properly as I described in the PR description.  Before, it looked like this:

![image](https://user-images.githubusercontent.com/25914066/170786017-5a598954-58f3-47f9-a93a-715ee6c10eae.png)

But hard coding it makes it display correctly:

![image](https://user-images.githubusercontent.com/25914066/170785607-d9a941c5-d938-4ef8-9150-046a9d27ef4d.png)

mskelton replied:
If you want, I could make constants for each of the icons used where each icon is just the actually glyph.  Then it could be referenced everywhere since it is just a string and not a lipgloss style.

dlvhdr replied:
Ah ok, yeah the constants are with the styles already. Cool...
You could make the glyphs a constant but maybe that's just overkill...
